### DISCUSSION TITLE
Github page shows only a part of the page but not the JS part

### DISCUSSION THREAD

**Discussion by Metanee97** [community/community#65468] [Other Feature Feedback, Questions, & Ideas] [answered]
### GITHUB PAGES

Question

### Body

Hello, 

I build a school project with html, css and JS. I've deployed it on github pages but the only things that show up is the element on the html. The JS part doesn't work but everything is OK on local (on my computer).

I'v already change all the relatives paths. And i add a .nojekyll file.

 I don't understand where it comes from. If someone can help me.
 
 My repo : https://github.com/Metanee97/OC-P6-Fisheye
 Github page: https://metanee97.github.io/OC-P6-Fisheye/

**Comment by proksh** [comment, marked as answer]
On your website, the problem seems to be with photograph.json. It is not returning the correct json file but a 404 page of Github (check in network tab of google console).

In `js/pages/index.js`, you are fetching this json like this -
```
async function getPhotographers() {
  return fetch('../data/photographers.json')
    .then(res => {
      return res.json();
    })
    .then(dataJSON => {
      //console.log({dataJSON})
      return dataJSON;
    });
}
```

Since `index.html` is in the root folder and `photographers.json` is directly under the `data` folder in the root, the fetch path should just be 'data/photographers.json'.

```
// get all datas
async function getPhotographers() {
  return fetch('data/photographers.json')  // Updated path
    .then(res => {
      if (!res.ok) {
        throw new Error('Network response was not ok');
      }
      return res.json();
    })
    .then(dataJSON => {
      //console.log({dataJSON})
      return dataJSON;
    });
}
```

This assumes that the web server serves files starting from the root directory where `index.html` resides. If you're using a development server, make sure the server's root directory is set to the project root.

**Reply by Metanee97** [reply]
Thank you so so so so so much!!! It's working now.

**Comment by Binary-Ninja-007** [comment]
Good.
